[IMP] Merged cycle counts docs and updated for 17.0 (#10685)
odoo/documentation: content/applications/inventory_and_mrp/inventory/warehouses_storage/inventory_management/count_products.rst

to reconcile the differences, and ensure that the recorded counts in the database match the actual
counts in the warehouse.

.. _inventory_management/count_products/inventory-adjustments-page:

Inventory Adjustments page
==========================

To view the *Inventory Adjustments* page, navigate to

> :menuselection:`Inventory app --> Operations --> Physical Inventory`.

Reviewer: This line still isn't correct. Can you take another look?

**Tips**
- commit suggestions to a batch commit that way they're applied as they're written; or
- copy the suggestion as it was written into your local copy of the doc so it drops in as written.
- always run `make review` before pushing up changes (we built this tool to catch line break issues like these 😉 )

> .. image:: count_products/inventory-adjustments-page.png

Reviewer 2: Will add this in new PR later!

> .. image:: count_products/inventory-adjustments-page.png

Reviewer: Resolving since will be addressed later.

to reconcile the differences, and ensure that the recorded counts in the database match the actual
counts in the warehouse.

> .. _inventory_management/count_products/inventory-adjustments-page:
> 

Reviewer 3: Small update to the custom anchor name to follow the convention:
> `.. _app_name/optional_parent_folder/section-name:`
```suggestion
.. _inventory/inventory-adjustments-page:

```

  can contain letters, numbers, or a combination of both.
- :guilabel:`UoM`: the *unit of measure* in which the product is measured. Unless otherwise
  specified (i.e., in :guilabel:`Pounds` or :guilabel:`Ounces`), the default :abbr:`UoM (Unit of
  Measure)` is :guilabel:`Units`.
- :guilabel:`Counted Quantity`: the real quantity counted during an inventory count. This field is
  left blank by default but can be changed, depending on if it matches the :guilabel:`On Hand
  Quantity` or not.
- :guilabel:`Difference`: the difference between the :guilabel:`On Hand Quantity` and
  :guilabel:`Counted Quantity`, once an inventory adjustment is made. The difference is
  automatically calculated after every inventory adjustment.

.. note::
   If a specific product has a quantity of more than `1.00` in stock, and more than one serial
   number (or lot number) is assigned, each uniquely-identified product is displayed on its own
   product line with its own lot/serial number, under the :guilabel:`Lot/Serial Number`
   column.

The fields below are optional and can be made visible by clicking the :icon:`oi-settings-adjust`

> :guilabel:`(settings)` icon in the top right of the :guilabel:`Inventory Adjustments` page.

Reviewer 3: Nitpick, the hyphenated top-right is preferred :)
```suggestion
:guilabel:`(settings)` icon in the top-right of the :guilabel:`Inventory Adjustments` page.
```

Reviewer 2: Since it's not modifying a noun, it's just top right in this case! :) (Zac and I hashed this one out already lol)

  value of this field is set to `0`.
- :guilabel:`Favorite`: adds a clickable star icon next to each product. Adding a favorite will
  bring the line item to the top of the dashboard.
- :guilabel:`Product Category`: the category assigned internally to a specific product. Unless
  otherwise specified (i.e., as :guilabel:`Consumable` or :guilabel:`Rental`), the default *Product
  Category* is set to :guilabel:`All`.
- :guilabel:`Expiration Date`: the date a product will expire, if it is a perishable good. Only
  products that are tracked using lots and serial numbers can be assigned expiration information.
- :guilabel:`Last Count Date`: the date on which the last inventory count occurred.
- :guilabel:`Available Quantity`: the quantity of a specific product that is currently available,
  based on any outstanding/uncompleted sales orders, purchase orders, or manufacturing orders that
  might change the available quantity once fulfilled.
- :guilabel:`On Hand Quantity`: the quantity of the product currently recorded in the database.
- :guilabel:`Accounting Date`: the date on which the adjustments will be accounted for in the Odoo

>   *Accounting* app.

Reviewer 3: App names should use bold formatting :)
```suggestion
  **Accounting** app.
```

- :guilabel:`Scheduled Date`: the date at which a count should be made. If not otherwise specified,
  this date defaults to the 31st of December of the current year.
- :guilabel:`User`: the person assigned to the count in the database. This can either be the person
  physically counting the inventory, or applying the count in the database.
- :guilabel:`Company`: the company whose database these inventory adjustments are being made on. The

>   company is listed in the top right corner of the database, next to the user currently logged in.

Reviewer 3: ```suggestion
  company is listed in the top-right corner of the database, next to the user currently logged in.
```

.. tip::
   Some columns are hidden by default. To reveal these columns, click the
   :icon:`oi-settings-adjust` :guilabel:`(settings)` icon to the far right of the form's top row,
   and select any desired column to reveal by clicking the checkbox next to that option.

.. seealso::
   - :doc:`Product type <../../product_management/configure/type>`

>    - :doc:`Tracking with lots, serial numbers, and expiration dates <../../product_management/product_tracking>`

Reviewer 3: Please break lines that go over 100 chars (if possible) :)
```suggestion
   - :doc:`Tracking with lots, serial numbers, and expiration dates
     <../../product_management/product_tracking>`
```

Create an inventory adjustment
------------------------------

> To create a new inventory adjustment from the :ref:`Inventory Adjustments
> <inventory_management/count_products/inventory-adjustments-page>` page, click :guilabel:`New`. Doing
> so creates a new, blank inventory adjustment line *at the bottom of the page*.

Reviewer 3: Update ref link to custom anchor
```suggestion
To create a new inventory adjustment from the :ref:`Inventory Adjustments
<inventory/inventory-adjustments-page>` page, click :guilabel:`New`. Doing so creates a new, blank
inventory adjustment line at the bottom of the page.
```

On this blank inventory adjustment line, define the :guilabel:`Location` where the product is
stored. This is necessary to ensure the accuracy of the inventory adjustment, as the same product
can be stored in multiple locations.

> Next, click the drop-down menu under the :guilabel:`Product`column, and select a product. If the

Reviewer 3: Add a space please :)
```suggestion
Next, click the drop-down menu under the :guilabel:`Product` column, and select a product. If the
```

==========================

To view the *Inventory Adjustments* page, navigate to :menuselection:`Inventory app --> Operations

> --> Physical Inventory`.

Reviewer 4: This line is great! But I'm gonna leave this comment here because I can't comment on lines 24 - 58. 

The existing documentation covers most fields, but in 17.0, they made some more column options.
![image](https://github.com/user-attachments/assets/49080014-b0cc-43e6-a5fa-9dd3c719e5fe)

Our documentation should also define what information each of these fields display. Can you do that for me, when implementing my suggestions? (:

For the sake of organizing more relevant information higher in the section, my suggestion is this:
- display the definitions of the columns available in the Inventory Adjustment table, first (you already have this :smile: )
- Then, in a note admonition block, write something like, "The fields below are optional fields that can be made visible by clicking the :icon:`oi-settings-adjust` :guilabel:`(settings)` icon in the top-right of the :guilabel:`Inventory Adjustments` page. 
- define each of the optional fields (let me know if you need help finding information about any of these fields! I suggest trying to look them up in the documentation first, and if you absolutely can't find anything, lmk :heart:)

.. note::
   If a specific product has a quantity of more than `1.00` in stock, and more than one serial
   number (or lot number) is assigned, each uniquely-identified product is displayed on its
   own product line with its own lot/serial number, displayed under the

>    :guilabel:`Lot/Serial Number` column.

Reviewer: This suggestion was not implemented correctly.
Please try again 🙏

The fields below are optional and can be made visible by clicking the
:icon:`oi-settings-adjust` :guilabel:`(settings)` icon in the top-right of the

> :guilabel:`Inventory Adjustments` page.

Reviewer: This suggestion was not implemented all the way:
"top right" gets a hyphen because it's a compound adjective, and it needs a noun after it since it's meant to qualify one.
Please try again.

- :guilabel:`Inventory Frequency (Days)`: the frequency, in days, for inventory counts. The default

>   value of this field is set to `0`.

Reviewer: Close! It's **Favorite** singular, as it's written on the UI (not **Favorites** plural, which is a different feature in the search bar).

  Quantity` or not.
- :guilabel:`Difference`: the difference between the :guilabel:`On Hand Quantity` and
  :guilabel:`Counted Quantity`, once an inventory adjustment is made. The difference is
  automatically calculated after every inventory adjustment.
- :guilabel:`Scheduled Date`: the date at which a count should be made. If not otherwise specified,
  this date defaults to the 31st of December of the current year.
- :guilabel:`User`: the person assigned to the count in the database. This can either be the person
  physically counting the inventory, or applying the count in the database.
- :guilabel:`Company`: the company whose database these inventory adjustments are being made on. The
  company is listed in the top right corner of the database, next to the user currently logged in.

.. tip::
   Some columns are hidden by default. To reveal these columns, click the
   :icon:`oi-settings-adjust` :guilabel:`(settings)` icon to the far right of the form's top row,
   and select any desired column to reveal by clicking the checkbox next to that option.

.. seealso::
   - :doc:`Product type <../../product_management/configure/type>`

>    - :doc:`Product tracking <../../product_management/product_tracking>`, including
>       using lots, serial numbers, and expiration dates

Reviewer 4: Something weird is going on with the formatting :cry: (see screenshot below)

Let's try making all of the text a link. But, if you have better ideas to make the text look good, Liz, feel free to try! 

![image](https://github.com/user-attachments/assets/b2309ae3-90da-4002-ae2a-59678d625df0)

```suggestion
   - :doc:`Product tracking, including using lots, serial numbers, and expiration dates <../../product_management/product_tracking>`
```

Or, changing the meaning a little but shortening the link text?
```suggestion
   - :doc:`Tracking lots, serials, and expiration dates <../../product_management/product_tracking>`
```

  :guilabel:`Counted Quantity`, once an inventory adjustment is made. The difference is
  automatically calculated after every inventory adjustment.
- :guilabel:`Scheduled Date`: the date at which a count should be made. If not otherwise specified,

>   this date defaults to the 31st of December of the current year.

Reviewer 4: commenting here since I can't comment on line 69, but in 16.0+, the "Create" button is changed to "New"
![image](https://github.com/user-attachments/assets/1d9fa3a2-09a6-47b3-ae21-0dbc04550705)

Also, we have yet to mention the "Relocate" button in this doc (it's new in 17.0!). This is also my bad, as I forgot to mention it in the task. 

The relocate button is a quick and easy way to move objects to another location. It happens when products end up in unexpected places and we need to record where it's actually at, or where it's supposed to be stored. Let me know if you have any additional questions about the Relocate feature! I'll let you decide where it's best to put this section (:

Reviewer 4: In line 61, there's a reference of the "Additional options" button. We no longer call it that, and instead just use :icon:`oi-settings-adjust` :guilabel:`(settings)` icon.

Reviewer 4: In line 109, we mention the "Apply button" at the far right of the page. While it's correct, I'd like you to modify it in your edit to include the icon. (: 
![image](https://github.com/user-attachments/assets/e7036717-7b94-4fa0-9a6d-2da5636fd098)

column.

.. image:: count_products/difference-column.png
   :alt: Difference column on inventory adjustments page.

At this stage, the count (:dfn:`inventory adjustment`) is recorded, but not yet applied. This means
that the quantity on hand before the adjustment has not yet been updated to match the new, real
counted quantity.

There are two ways to apply the new inventory adjustment. The first way is to click the
:icon:`fa-save` :guilabel:`Apply` button on the line at the far right of the page. The second way is
to click the checkbox on the far left of the line. Doing so reveals new button options at the top of
the page, one of which is an :guilabel:`Apply` button. Clicking this button instead causes an

> :guilabel:`Inventory Adjustment Reference / Reason` pop-up window to appear.

Reviewer 3: This may have been covered previously, is does this pop-up change to "Reference" or "Reason" on a condition?

Either way, the pop-up's name is only "Inventory Adjustment"
```suggestion
:guilabel:`Inventory Adjustment` pop-up window to appear.
```

.. image:: count_products/apply-inventory-adjustment.png
   :alt: Apply all option applies the inventory adjustment once a reason is specified.

From this pop-up menu, a reference or reason can be assigned to the inventory adjustment. By

> default, the :guilabel:`Inventory Reference / Reason` field is pre-populated with the date the
> adjustment is being made on, but can be changed to reflect whatever reference or reason is desired.

Reviewer 3: Please disregard if the field name does change conditionally
```suggestion
default, the :guilabel:`Inventory Reason` field is pre-populated with the date the adjustment is
being made on, but can be changed to reflect whatever reference or reason is desired.
```

column.

.. image:: count_products/difference-column.png
   :alt: Difference column on inventory adjustments page.

At this stage, the count (:dfn:`inventory adjustment`) is recorded, but not yet applied. This means
that the quantity on hand before the adjustment has not yet been updated to match the new, real
counted quantity.

There are two ways to apply the new inventory adjustment. The first way is to click the
:icon:`fa-save` :guilabel:`Apply` button on the line at the far right of the page. The second way is
to click the checkbox on the far left of the line. Doing so reveals new button options at the top of
the page, one of which is an :guilabel:`Apply` button. Clicking this button instead causes an
:guilabel:`Inventory Adjustment Reference / Reason` pop-up window to appear.

> .. image:: count_products/apply-inventory-adjustment.png

Reviewer 2: Will add this in new PR later!

> .. image:: count_products/apply-inventory-adjustment.png

Reviewer: Resolving since will be addressed later.

   :alt: Apply all option applies the inventory adjustment once a reason is specified.

From this pop-up menu, a reference or reason can be assigned to the inventory adjustment. By
default, the :guilabel:`Inventory Reference / Reason` field is pre-populated with the date the
adjustment is being made on, but can be changed to reflect whatever reference or reason is desired.

Once ready, click :guilabel:`Apply` to apply the inventory adjustment.

.. note::
   Applying an inventory adjustment simultaneously creates a
   :doc:`stock move line (SML) <../reporting/moves_history>` in the *Moves History* report for

>    traceability.

Reviewer: This suggestion was not implemented. Please try again.

>    traceability.

Reviewer 2: Didn't realize the suggestion was a preferred way of doing it.  so I had initially left it as-is so the link could remain highlighted in VSCode (I know it doesn't mean the link it broken if it's not highlighted, but as a writer, it makes it easier to look for and format doc/ref/menu selection links when they're highlighted).

However, Sam filled me in on the technical reason behind why this is, so I have fixed this and will look out for it moving forward!

.. tip::
   Sometimes products end up in unexpected or incorrect locations. To quickly move products to a
   different storage location, click the :guilabel:`Relocate` button at the top of the
   :ref:`Inventory Adjustments <inventory_management/count_products/inventory-adjustments-page>`

>    page.

Reviewer 2: Will add this in new PR later!

Count products
==============

Counting products is a recurring activity in a warehouse. Once a count is complete, go to
:menuselection:`Inventory app --> Operations --> Physical Inventory` to update the

> :guilabel:`Counted Quantity` column for each product line.

Reviewer 2: @StraubCreative @Felicious I agree with that assessment, Zac! Admittedly I was trying to quickly incorporate Felicia's initial and second rounds of feedback on this doc without diving deeply into the sections that weren't commented on as much. 

I will follow up with a second PR to fix this section + add screenshots.

> :guilabel:`Counted Quantity` column for each product line.

Reviewer: Sounds good, resolving for now then 👍

counted quantity.

There are two ways to apply the new inventory adjustment. The first way is to click the
:icon:`fa-save` :guilabel:`Apply` button on the line at the far right of the page. The second way is
to click the checkbox on the far left of the line. Doing so reveals new button options at the top of
the page, one of which is an :guilabel:`Apply` button. Clicking this button instead causes an
:guilabel:`Inventory Adjustment Reference / Reason` pop-up window to appear.

> .. image:: count_products/apply-inventory-adjustment.png

Reviewer 4: Great idea to add an image here!

>    :align: center

Reviewer 4: Slightly getting ahead of myself here since this is something Sam catches in technical reviews, but we're no longer using the `align` tag for images. 

It's not required for this stage of review, but can you please remove all instances of `align` by the next review stage? Thank you :blush: 

Not sure if you already knew, but there's a find and replace shortcut in VSCode!
- CTRL + F
- Click the ">" arrow icon in the pop-up to reveal a drop-down menu
- Leave the 'Replace' field blank and click the 'Enter' key on your keyboard to find and remove that word

![image](https://github.com/user-attachments/assets/fb9d3fae-48d6-47e7-925d-c4bedb7b5270)

Reviewer 2: Thank you!! I was going to delete them, but since Kara had left them (didn't realize this was her first PR until you'd pointed that out :) ), I was wondering if maybe this convention was supposed to be left for 17.0 docs. 

Thanks so much for this pro tip with the find + replace function in VSCode - so helpful! :D

default, the :guilabel:`Inventory Reference / Reason` field is pre-populated with the date the
adjustment is being made on, but can be changed to reflect whatever reference or reason is desired.

Once ready, click :guilabel:`Apply` to apply the inventory adjustment.

.. note::
   Applying an inventory adjustment simultaneously creates a
   :doc:`stock move line (SML) <../reporting/moves_history>` in the *Moves History* report for
   traceability.

.. tip::
   Sometimes products end up in unexpected or incorrect locations. To quickly move products to a
   different storage location, click the :guilabel:`Relocate` button at the top of the
   :ref:`Inventory Adjustments <inventory_management/count_products/inventory-adjustments-page>`
   page.

Count products
==============

Counting products is a recurring activity in a warehouse. Once a count is complete, go to
:menuselection:`Inventory app --> Operations --> Physical Inventory` to update the
:guilabel:`Counted Quantity` column for each product line.

To print a PDF of a count sheet, add products to the count sheet by ticking their checkboxes. Click

> the :icon:`fa-print` :guilabel:`(printer icon)`, then select :guilabel:`Count sheet` to

Reviewer 4: When an icon accompanies text, we don't need to describe the icon in guilabels (: 
```suggestion
the :icon:`fa-print` :guilabel:`Print` button at the top, then select :guilabel:`Count sheet` to
```

Reviewer 2: Hmm, I think I see what you're saying. I was trying to follow this first descriptive icon style from the RST guidelines: 
![image](https://github.com/user-attachments/assets/a4988783-1487-41f8-b37b-896c161237b4)

Reviewer 2: Ahhh, I went back to the UI, and I see what you mean! I hadn't realized it said 'Print' next to the icon. Thanks for pointing that out! :)

download a PDF that details the selected products.

On each product line, identify whether the value in the :guilabel:`On Hand Quantity` column recorded
in the database matches the newly-counted value. If the recorded value and the counted value do

> match, select the product using the checkbox, and then select the :guilabel:`Actions` button, then
> :guilabel:`Set to quantity on hand`.

Reviewer 4: Minor edit to remove the use of double "then"

```suggestion
match, select the product using the checkbox, click the :guilabel:`Actions` button, then
:guilabel:`Set to quantity on hand`.
```

==============

Counting products is a recurring activity in a warehouse. Once a count is complete, go to
:menuselection:`Inventory app --> Operations --> Physical Inventory` to update the
:guilabel:`Counted Quantity` column for each product line.

To print a PDF of a count sheet, add products to the count sheet by ticking their checkboxes. Click

> :guilabel:`Count sheet` to download a PDF that contains the selected products.

Reviewer 4: The count sheet button is not available by default; you have to click the Print button to get to it, first (:

![image](https://github.com/user-attachments/assets/f8392d5c-b687-40a1-a3e3-456ce1d8abdf)

Please tweak this line a little bit to have the documentation reflect *precisely* how things appear in the UI (:

Also! This a great opportunity to use the printer icon
```
:icon:`fa-print me-1`
```

On each product line, identify whether the value in the :guilabel:`On Hand Quantity` column recorded
in the database matches the newly-counted value. If the recorded value and the counted value do
match, select the product using the checkbox, and then select

> :menuselection:`Actions --> Set to quantity on hand`.

Reviewer 4: This is our team's specific style, but we only use the menuselection tag when navigating menus 
![image](https://github.com/user-attachments/assets/30680b5e-4a37-417c-b3fd-5708472efb15)

Actions is a button that has "drop-down menu options". Very minor distinction, I know. :see_no_evil: 

```suggestion
the :guilabel:`Actions` button, then :guilabel:`Set to quantity on hand`.
```

   Applying an inventory adjustment simultaneously creates a :doc:`stock move line (SML)
   <../reporting/moves_history>` in the *Moves History* report for traceability.

.. tip::
   Sometimes products end up in unexpected or incorrect locations. To quickly move a product to a
   different storage location, check the box next to the product, and click the :guilabel:`Relocate`

>    button at the top of the :ref:`Inventory Adjustments
>    <inventory_management/count_products/inventory-adjustments-page>` page.

Reviewer 3: Update ref to custom anchor
```suggestion
   button at the top of the :ref:`Inventory Adjustments <inventory/inventory-adjustments-page>`
   page.
```

away from the line. Doing so saves the adjustment, and automatically adjusts the value in the
:guilabel:`Difference` column.

Subsequently, once applied, a move with the difference between the :guilabel:`On Hand Quantity` and
the :guilabel:`Counted Quantity` is recorded in the product's inventory adjustment history.

.. image:: count_products/history-inventory-adjustments.png
   :align: center
   :alt: Inventory Adjustments History dashboard detailing a list of prior product moves.

The :guilabel:`Actions` menu appears when one or more products' checkboxes are selected. The
:guilabel:`Actions` menu includes the option to :guilabel:`Set to quantity on hand`, which sets the
selected products' :guilabel:`Counted Quantity` to the :guilabel:`On Hand Quantity`, and

> :guilabel:`Set to 0`, which sets the selected products' :guilabel:`Counted Quantity` to zero.

Reviewer 4: oh you did it perfectly here!!! :heart_eyes:

Finally, once ready, click :guilabel:`Confirm` to request the count.

.. image:: count_products/count-popup.png
   :alt: Request a count popup on inventory adjustments page.

.. important::

>    In the Odoo *Barcode* app, users can only view inventory counts that are assigned to **them**,

Reviewer 3: Please bold app names :)
```suggestion
   In the Odoo **Barcode** app, users can only view inventory counts that are assigned to **them**,
```

Finally, once ready, click :guilabel:`Confirm` to request the count.

.. image:: count_products/count-popup.png

>    :alt: Request a count popup on inventory adjustments page.

Reviewer 4: Love that you already removed "align" here!
